WordPress/Classes/ViewRelated/Post/PostEditorNavigationBarManager.swift

```diff
@@ -0,0 +1,179 @@
+        cancelItem.accessibilityIdentifier = "Close"
+        return cancelItem
+    }()
+
+
+    /// NavigationBar's Blog Picker Button
+    ///
+    private lazy var blogPickerBarButtonItem: UIBarButtonItem = {
+        let pickerItem = UIBarButtonItem(customView: self.blogPickerButton)
+        pickerItem.accessibilityLabel = NSLocalizedString("Switch Blog", comment: "Action button to switch the blog to which you'll be posting")
+        return pickerItem
+    }()
+
+    /// Media Uploading Status Button
+    ///
+    private lazy var mediaUploadingBarButtonItem: UIBarButtonItem = {
+        let barButton = UIBarButtonItem(customView: self.mediaUploadingButton)
+        barButton.accessibilityLabel = NSLocalizedString("Media Uploading", comment: "Message to indicate progress of uploading media to server")
+        return barButton
+    }()
+
+    /// Publish Button
+    private(set) lazy var publishBarButtonItem: UIBarButtonItem = {
+        let button = UIBarButtonItem(customView: self.publishButton)
+
+        return button
+    }()
+
+    /// NavigationBar's More Button
+    ///
+    lazy var moreBarButtonItem: UIBarButtonItem = {
+        let moreItem = UIBarButtonItem(customView: self.moreButton)
+        return moreItem
+    }()
+
+    // MARK: - Selectors
+
+    @objc private func closeWasPressed(sender: UIButton) {
+        delegate?.navigationBarManager(self, closeWasPressed: sender)
+    }
+
+    @objc private func moreWasPressed(sender: UIButton) {
+        delegate?.navigationBarManager(self, moreWasPressed: sender)
+    }
+
+    @objc private func blogPickerWasPressed(sender: UIButton) {
+        delegate?.navigationBarManager(self, blogPickerWasPressed: sender)
+    }
+
+    @objc private func publishButtonTapped(sender: UIButton) {
+        delegate?.navigationBarManager(self, publishButtonWasPressed: sender)
+    }
+
+    @objc private func displayCancelMediaUploads(sender: UIButton) {
+        delegate?.navigationBarManager(self, displayCancelMediaUploads: sender)
+    }
+
+    // MARK: - Public
+
+    var leftBarButtonItems: [UIBarButtonItem] {
+        return [separatorButtonItem, closeBarButtonItem, blogPickerBarButtonItem]
+    }
+
+    var uploadingMediaLeftBarButtonItems: [UIBarButtonItem] {
+        return [separatorButtonItem, closeBarButtonItem, mediaUploadingBarButtonItem]
+    }
+
+    var rightBarButtonItems: [UIBarButtonItem] {
+        return [moreBarButtonItem, publishBarButtonItem, separatorButtonItem]
+    }
+
+    func reloadPublishButton() {
+        publishButton.setTitle(delegate?.publishButtonText ?? "", for: .normal)
+        publishButton.isEnabled = delegate?.isPublishButtonEnabled ?? true
+    }
+
+    func reloadBlogPickerButton(with title: String, enabled: Bool) {
+
+        let titleText = NSAttributedString(string: title, attributes: [.font: AztecPostViewController.Fonts.blogPicker])
```

Comment:
Same as above... considering the changes in this PR `AztecPostViewController.Fonts.blogPicker` should be moved to this class.

The rationale is that this class shouldn't really care or know about `AztecPostViewController` specifically.